docs/dev/dev_adr.md

```diff
@@ -0,0 +1,111 @@
+# Architecture Decision Records
+
+The intention is to document deviations from a standard Model View Controller (MVC) design.
+
+## Pivoting Compliance View
+
+The view that was preferred for compliance would be devices on y-axis but the features on the x-axis. However, the x-axis (features) cannot be known when building the Django model. The model ends up looking like:
+
+| Device   | feature | Compliance |
+| -------- | ------- | ---------- |
+| nyc-rt01 | aaa     | True       |
+| nyc-rt01 | ntp     | False      |
+| nyc-rt01 | dns     | False      |
+| nyc-rt02 | aaa     | True       |
+| nyc-rt02 | dns     | False      |
+
+The expected view required expected is something like:
+
+| Device   | aaa   | dns   | ntp   |
+| -------- | ----- | ----- | ----- |
+| nyc-rt01 | True  | False | False |
+| nyc-rt02 | True  |       | False |
+
+In order to accommodate this, `django-pivot` is used, which greatly simplifies building the query. However, `django-pivot` requires the ability to "count" vs a boolean. Because of that, there is a "shadow" field created that sets to 0 if False, and 1 if True. This is enforced on the `save` method of the `ConfigCompliance` model.
```

Comment:
```suggestion
In order to accommodate this, `django-pivot` is used, which greatly simplifies building the query. However, `django-pivot` requires the ability to "count" versus a boolean. Because of that, there is a "shadow" field created that is set to 0 if False, and 1 if True. This is enforced on the `save` method of the `ConfigCompliance` model.
```